frontend/apps/reader/modules/readerlink.lua

```diff
@@ -91,6 +92,138 @@ function ReaderLink:init()
 
     -- delegate gesture listener to readerui, NOP our own
     self.ges_events = nil
+
+    -- Set up buttons for alternative external link handling methods
+    self._external_buttons["01_copy"] = function(this, link_url)
+        return {
+            text = _("Copy"),
+            callback = function()
+                UIManager:close(this.dialog)
+            end,
+        }
+    end
+
+    self._external_buttons["02_qrcode"] = function(this, link_url)
+        return {
+            text = _("Show QR code"),
+            callback = function()
+                UIManager:close(this.dialog)
+                UIManager:show(QRMessage:new{
+                        text = link_url,
```

Comment:
You really like to mess our nice "4 spaces per level" indentation :/ Or it is while moving it around and fixing manually each line indentation (and your text editor is not a good friend).
(Others below).

Reply:
I'm sorry :( I'm using Emacs and I don't really know how to do setup up the automatic indents properly.